guacamole-docker/bin/start.sh

```diff
@@ -90,15 +90,21 @@ set_optional_property() {
 ## instead of a linked container.
 ##
 associate_mysql() {
-
-    # Use linked container if specified
-    if [ -n "$MYSQL_NAME" ]; then
-        MYSQL_HOSTNAME="$MYSQL_PORT_3306_TCP_ADDR"
-        MYSQL_PORT="$MYSQL_PORT_3306_TCP_PORT"
-    fi
-
-    # Use default port if none specified
-    MYSQL_PORT="${MYSQL_PORT-3306}"
+	: ${MYSQL_HOSTNAME:=mysql}
```

Comment:
Using `MYSQL_PORT_3306_TCP_ADDR` and `MYSQL_PORT_3306_TCP_PORT` is deprecated as of [Link environment variables reference](https://docs.docker.com/compose/link-env-deprecated/).
Linked services are reachable by the service name or its alias. In this case the alias of the database service is `mysql`. Please see [Links](https://docs.docker.com/compose/compose-file/#links).
Docker is either adding a `Hosts`entry for the linked docker image, newer docker versions do have a build in DNS server.